share/mk/bsd.cpu.mk

```diff
@@ -336,7 +336,9 @@ MACHINE_CPU += riscv
 . if ${MACHINE_CPU:Mcheri}
 CFLAGS+=	-march=morello
 CFLAGS+=	-Xclang -morello-vararg=new -Xclang -morello-bounded-memargs
+CFLAGS+=	-cheri-codeptr-relocs
```

Comment:
We could instead make this optional on a new MK_FOO for now and then we can merge this into `dev` without requiring the compiler to be updated and later toggle the default for MK_FOO?  (Or make it a COMPILER_FEATURE perhaps)?